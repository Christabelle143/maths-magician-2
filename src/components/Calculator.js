import React from 'react';

class Calculator extends React.PureComponent {
  render() {
    return (
      <div className="calculator">
        <input type="number" className="calc-input" />
        <table className="calc-buttons">
          <tr className="tr">
            <td className="button1">AC</td>
            <td className="button1">+/-</td>
            <td className="button1">%</td>
            <td className="button2"><span>&#247;</span></td>
          </tr>
          <tr>
            <td className="button1">7</td>
            <td className="button1">8</td>
            <td className="button1">9</td>
            <td className="button2">x</td>
          </tr>
          <tr className="tr">
            <td className="button1">4</td>
            <td className="button1">5</td>
            <td className="button1">6</td>
            <td className="button2">-</td>
          </tr>
          <tr className="tr">
            <td className="button1">1</td>
            <td className="button1">2</td>
            <td className="button1">3</td>
            <td className="button2">+</td>
          </tr>
          <tr className="tr">
            <td className="button1" colSpan="2">0</td>
            <td className="button1">.</td>
            <td className="button2">=</td>
          </tr>
        </table>
      </div>
    );
  }
}
export default Calculator;
