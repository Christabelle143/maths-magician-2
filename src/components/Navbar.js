import { Link } from 'react-router-dom';

const Navbar = () => (
  <>
    <nav className="navbar">
      <div className="mainheading">
        <h1> Math Magicians </h1>
      </div>
      <div className="links">
        <Link to="/">Home</Link>
        <Link to="/Calculator">Calculator</Link>
        <Link to="/Quote">Quote</Link>
      </div>
    </nav>

  </>
);
export default Navbar;
