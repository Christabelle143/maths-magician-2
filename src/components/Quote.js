const Quote = () => (
  <div className="quote">
    <p>
      <q className="quotes">
        {' '}
        Mathmatics is not about numbers,equations, computations,
        or algorithms: it is about understanding.
        <span className="author">-William Paul Thurston</span>
      </q>
    </p>
  </div>
);

export default Quote;
