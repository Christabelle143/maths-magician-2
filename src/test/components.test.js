import React from 'react';
import renderer from 'react-test-renderer';
import { Router } from 'react-router-dom';
import Calculator from '../components/Calculator';
import Navbar from '../components/Navbar';
import Home from '../components/Home';
import Quote from '../components/Quote';

it('should render home component correctly', () => {
  const home = renderer
    .create(<Home />)
    .toJSON();
  expect(home).toMatchSnapshot();
});
it('should render quote component correctly', () => {
  const quote = renderer
    .create(<Quote />)
    .toJSON();
  expect(quote).toMatchSnapshot();
});
it('should render Calculator component correctly', () => {
  const calculate = renderer
    .create(<Calculator />)
    .toJSON();
  expect(calculate).toMatchSnapshot();
});
it('should render Navbar component correctly', () => {
  <Router>
    const navbar = renderer
    .create(
    <Navbar />
    )
    .toJSON();
    expect(navbar).toMatchSnapshot();
  </Router>;
});
