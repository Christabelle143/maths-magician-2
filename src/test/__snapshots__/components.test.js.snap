// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Calculator component correctly 1`] = `
<div
  className="maths"
>
  <div
    className="math"
  >
    <h2>
      let\`s do some math!
    </h2>
  </div>
  <div
    className="calculator-container"
  >
    <div
      className="calc-input"
    >
      <span
        className="result"
      >
        0
      </span>
    </div>
    <div
      className="buttons"
    >
      <button
        className="bg-1"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="bg-1"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="bg-1"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="bg-1 active"
        onClick={[Function]}
        type="button"
      >
        <span>
          รท
        </span>
      </button>
      <button
        className="bg-1"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="bg-1"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="bg-1"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="bg-1 active"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        className="bg-1"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="bg-1"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="bg-1"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="bg-1 active"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        className="bg-1"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="bg-1"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="bg-1"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="bg-1 active"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="zero"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="bg-1"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="bg-1 active"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;

exports[`should render home component correctly 1`] = `
<div>
  <div
    className="homepage"
  >
    <h3>
      Welcome to our page!
    </h3>
  </div>
  <p>
    Lorem ipsum dolor sit amet consectetur, adipisicing elit. Earum ipsum tempore, ea soluta tenetur expedita dignissimos similique deleniti doloribus fugiat! Necessitatibus, sint doloribus deserunt voluptates impedit cumque alias, aperiam illo dolore explicabo iure culpa distinctio inventore perspiciatis sunt ipsam atque.Lorem ipsum dolor sit amet consectetur, adipi sicing elit. Earum ipsum tempore, ea soluta tenetur expedita dignissimos similique deleniti doloribus fugiat! Necessitatibus, sint doloribus deserunt voluptates impedit cumque alias, aperiam illo dolore explicabo iure culpa distinctio inventore perspiciatis sunt ipsam a tque.
  </p>
  <p>
    Lorem ipsum dolor sit amet consectetur, adipisicing elit. Earum ipsum tempore, ea soluta tenetur expedita dignissimos similique deleniti doloribus fugiat! Necessitatibus, sint doloribus deserunt voluptates impedit cumque alias, aperiam illo dolore explicabo iure culpa distinctio inventore perspiciatis sunt ipsam atque.Lorem ipsum dolor sit amet consectetur, adipi sicing elit. Earum ipsum tempore, ea soluta tenetur expedita dignissimos similique deleniti doloribus fugiat! Necessitatibus, sint doloribus deserunt voluptates impedit cumque alias, aperiam illo dolore explicabo iure culpa distinctio inventore perspiciatis sunt ipsam a tque.
  </p>
</div>
`;

exports[`should render quote component correctly 1`] = `
<div
  className="quote"
>
  <p>
    <q
      className="quotes"
    >
       
      Mathmatics is not about numbers,equations, computations, or algorithms: it is about understanding.
      <span
        className="author"
      >
        -William Paul Thurston
      </span>
    </q>
  </p>
</div>
`;
